db/data_listeners.hh

```diff
@@ -45,6 +54,16 @@ public:
             const query::partition_slice& slice, flat_mutation_reader_v2&& rd) {
         return std::move(rd);
     }
+
+    virtual bool set_sampling_probability(double p);
+
+    virtual double get_sampling_probability() const {
+        return _sampling_probability;
+    }
+
+    virtual bool sample_next_query() {
+        return _normalized_sampling_probability != 0 && _gen() < _normalized_sampling_probability;
+    }
```

Comment:
Why are these virtual?

Reply:
I thought it would make sense to make them virtual in order to allow other derived data_listeners that we might create in the future to override these methods.

Reply:
I think you can drop the virtual. Those future users (if they materialize), can re-add it themselves. Until then, no need to pay the cost of a virtual call (however minuscule that might be).

Reply:
Done